pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/realtime/PinotLLCRealtimeSegmentManager.java

```diff
@@ -629,7 +629,8 @@ protected SegmentMetadataImpl extractSegmentMetadata(final String rawTableName,
           .unTarOneFile(new FileInputStream(segFile), V1Constants.MetadataKeys.METADATA_FILE_NAME);
       metadataPath = FileSystems.getDefault().getPath(baseDir, segmentNameStr + ".metadata");
       Files.copy(is, metadataPath);
-      segmentMetadata = new SegmentMetadataImpl(new File(metadataPath.toString()));
+      // TODO: crc is not loaded because it's inside creation.metadata file
+      segmentMetadata = SegmentMetadataImpl.fromMetadataFile(new File(metadataPath.toString()));
```

Comment:
This is not a test... I will remove the new static factory method and uniform the behavior (always load from a directory)